### DISCUSSION TITLE
Random 500 errors on production server. Static app is still fetching data ?

### DISCUSSION THREAD

**Discussion by atissedrescaleway** [vercel/next.js#70073] [Help]
### Summary

TL;DR : App is build statically with `getStaticProps`, still tries to fetch at request time, fails since API is protected under internal infra.

**Our stack :** 
- We are using Next.js (`v14.2.5`) with pages router for our production website. 
- We're self-hosting the app, on public serverless containers. 
- All of the pages are exclusively built using **getStaticProps**.
- The build of the app is made in our CI, in our internal infra, and requests data for the pages to an **API that's isn't accessible publicly**.
- We use the `output: standalone` option

**The issue** :

Occasionally, we notice in the logs some 500 Internal Server Error. These logs says `Error: Fetch failed..`.

--> Indeed, the fetch is logically failing because the public serverless container does not have access to the internal API where we have our data.

**But why is the app is even fetching data in the first place**, since none of our pages are using `getServerSideProps`? 

--> That's the thing we don't understand. We're building the site statically so it doesn't need no fetch anything at request time.
